### ISSUE TITLE
False positive in DuplicateMethods

### ISSUE THREAD

**Issue by bquorning** [rubocop/rubocop#15010]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/DuplicateMethods

### Steps to reproduce the problem

Run Rubocop on the following code:
```rb
Foo.prepend(
  Module.new { def name; end }
)
Bar.prepend(
  Module.new { def name; end }
)
```

### Expected behavior

The same method being defined on two different anonymous Modules, which are included/extended/prepended onto different classes, should not be regarded as a duplicate.

### Actual behavior

```
Lint/DuplicateMethods: Method Object#name is defined at both file.rb:2 and file.rb:5.
  Module.new { def name; end }
               ^^^^^^^^
```

### RuboCop version

Any version after 9c70cee9666a9ab086d5d4372cf5781324825b75 / #14996.
